src/check.h.in

```diff
@@ -659,6 +660,117 @@ CK_DLL_EXP void CK_EXPORT _ck_assert_failed(const char *file, int line,
  */
 #define ck_assert_uint_ge(X, Y) _ck_assert_uint(X, >=, Y)
 
+/* Floating point number comparison macros with improved output
+ * compared to ck_assert(). */
+/* OP may be any comparison operator. */
+#define _ck_assert_floating(X, OP, Y) do { \
+  long double _ck_x = (X); \
+  long double _ck_y = (Y); \
+  ck_assert_msg(_ck_x OP _ck_y, "Assertion '%s' failed: %s == %.6Lf, %s == %.6Lf", #X" "#OP" "#Y, #X, _ck_x, #Y, _ck_y); \
+} while (0)
+
+/**
+ * Check two floating point numbers to determine if X==Y
+ *
+ * If not X==Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_eq(X, Y) _ck_assert_floating(X, ==, Y)
```

Comment:
You may not want to offer a way to use the == operator on two floating point numbers. Because floating point math is not associative and may follow different rounding models, the outcome of a computation may not be exactly what one expects. The ck_assert_floating_eq_prec() call below is a much better option. I would request that this and the ck_assert_floating_ne() methods be removed.